tests/tests/tier0/bluechi-is-online-system-wait/test_bluechi_is_online_system_wait.py

```diff
@@ -0,0 +1,210 @@
+AGENT_TWO = "agent-two"
+IMMEDIATE_RETURN_TIMEOUT_MS = get_test_env_value_int("IMMEDIATE_RETURN_TIMEOUT", 1000)
+WAIT_PARAM_VALUE_MS = get_test_env_value_int("WAIT_PARAM_VALUE", 5000)
+SLEEP_DURATION = get_test_env_value_int("SLEEP_DURATION", 2)
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_system(
+    bluechi_is_online: BluechiIsOnline,
+    wait_time: int,
+    future: ResultFuture,
+):
+    future.result = bluechi_is_online.system_is_online(
+        wait_time,
+    )
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    # Test 1: Both agents and controller are online
+    LOGGER.debug("Testing with both agents and controller online.")
+    system_result_test_one = ResultFuture()
+    system_thread_wait = threading.Thread(
+        target=check_system,
+        args=(
+            ctrl.bluechi_is_online,
+            IMMEDIATE_RETURN_TIMEOUT_MS,
+            system_result_test_one,
+        ),
+    )
+    system_thread_wait.start()
+    system_thread_wait.join()
+    assert (
+        system_result_test_one.result
+    ), "System wait command should not produce output when system is running."
+
+    # Test 2: Agent-one offline
+    LOGGER.debug("Starting test number 2 - agent one should remain offline.")
+    LOGGER.debug("Stopping agent-one.")
+    nodes[AGENT_ONE].systemctl.stop_unit("bluechi-agent")
+    assert nodes[AGENT_ONE].wait_for_unit_state_to_be("bluechi-agent", "inactive")
+
+    start_time = time.time()
+    system_result_test_two = ResultFuture()
+    system_thread = threading.Thread(
+        target=check_system,
+        args=(
+            ctrl.bluechi_is_online,
+            IMMEDIATE_RETURN_TIMEOUT_MS,
+            system_result_test_two,
+        ),
+    )
+    system_thread.start()
+    try:
+        with Timeout(SLEEP_DURATION, "Timeout while waiting for system."):
+            system_thread.join()
+    except TimeoutError:
+        LOGGER.debug(
+            "Timeout reached while waiting for system. Attempting to terminate."
+        )
+
+    assert (
+        system_result_test_two.result is not None and not system_result_test_two.result
+    ), f"Expected bluechi-is-online system with --wait={IMMEDIATE_RETURN_TIMEOUT_MS} to return an error"
+    assert (
+        time.time() - start_time > IMMEDIATE_RETURN_TIMEOUT_MS / 1000
+    ), f"Expected around {IMMEDIATE_RETURN_TIMEOUT_MS / 1000} seconds for bluechi-is-online to exit"
+
+    # Bring agent-one back online
+    LOGGER.debug("Starting agent-one.")
+    nodes[AGENT_ONE].systemctl.start_unit("bluechi-agent")
+    assert nodes[AGENT_ONE].wait_for_unit_state_to_be("bluechi-agent", "active")
+
+    # Test 3: Agent-two offline
+    LOGGER.debug("Stopping agent-two.")
+    nodes[AGENT_TWO].systemctl.stop_unit("bluechi-agent")
+    assert nodes[AGENT_TWO].wait_for_unit_state_to_be("bluechi-agent", "inactive")
+
+    start_time = time.time()
```

Comment:
Same as above, shouldn't the test be similar to https://github.com/eclipse-bluechi/bluechi/blob/main/tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py#L53 ?